### PR TITLE
Implement Authorization Rules with CanCanCan

### PR DESCRIPTION
**Pull request by lily4178993**
## Description

This pull introduce several updates to enhance the project's security and permissions. The key changes include:

1. **Installed CanCanCan:**
   - Integrated CanCanCan into the project to manage and enforce authorization rules.

2. **Added 'role' Column to Users Table:**
   - Created a migration to add a 'role' column to the users table.

3. **Post Deleting Functionality:**
   - Users can now delete their own posts or any post if they have an 'admin' role.
   - Implemented the post deleting functionality.
   - Added a "Delete" button to the view, visible only to authorized users.

4. **Comment Deleting Functionality:**
   - Users can now delete their own comments or any comment if they have an 'admin' role.
   - Implemented the comment deleting functionality.
   - Added a "Delete" button to the view, visible only to authorized users.

## Checklist

- [X] CanCanCan is properly configured.
- [X] The 'role' column migration has been applied successfully.
- [X] Users can delete their own posts/comments.
- [X] Users with an 'admin' role can delete any post/comment.
- [X] "Delete" button is visible only to authorized users.

---
**Thank you for your time and attention to this pull request.**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
          :recoverable, :rememberable, :validatable, :confirmable
-  has_many :posts, foreign_key: 'author_id'
-  has_many :comments
-  has_many :likes
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
 
   # Validation: Name must not be blank
   validates :name, presence: true
```
Comment (mukhammadrizooff) [inline app/models/user.rb:9]: - [ ] [REQUIRED] Good job so far! Kindly, you should create your `user.admin?` method inside the user model. like below, please implement it🙏🏼 

```ruby
...
   def admin?
     role == 'admin' ? true : false
   end
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @TEAM, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (lily4178993) [reply app/models/user.rb:9]: @mukhammadrizooff Thank you for letting me know, I've already made the changes

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**